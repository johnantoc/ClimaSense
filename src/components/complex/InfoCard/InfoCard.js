/**
 * Info Card Component.
 */
import React, { useMemo } from "react";
import { View, FlatList } from "react-native";
import { useSelector } from "react-redux";

import RichText from "../../basic/RichText/RichText";
import ButtonText from "../../basic/ButtonText/ButtonText";
import DayItem from "../../basic/DayItem/DayItem";
import InfoCardStyles from "./InfoCard.style";

const InfoCard = ({ navigation }) => {
  const {
    weather: { dailyData },
    settings: { tempUnitF },
  } = useSelector((state) => {
    return {
      weather: state.weatherData,
      settings: state.settings,
    };
  });

  const renderItem = ({ item }) => (
    <DayItem item={item} tempUnitF={tempUnitF} />
  );
  const renderFunction = useMemo(() => renderItem, [dailyData, tempUnitF]);

  return (
    <View style={[InfoCardStyles.container]}>
      <View style={[InfoCardStyles.mainArea]}>
        <View style={[InfoCardStyles.headingContainer]}>
          <RichText text="Next 7 Days" styles={InfoCardStyles.heading} />
          <View style={InfoCardStyles.buttonContainer}>
            <ButtonText
              textStyles={InfoCardStyles.details}
              title="Detailed View"
              onPressHandler={() => navigation.navigate("Details")}
            />
          </View>
        </View>
        <View style={[InfoCardStyles.listContainer]}>
          <FlatList
            data={dailyData}
            renderItem={renderFunction}
            keyExtractor={(item) => item.id}
          />
        </View>
      </View>
    </View>
  );
};

export default InfoCard;
