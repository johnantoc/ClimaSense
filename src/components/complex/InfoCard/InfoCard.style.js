import { StyleSheet } from "react-native";

const InfoCardStyles = StyleSheet.create({
  container: {
    flex: 0.4,
    backgroundColor: "transparent",
  },
  mainArea: {
    flex: 1,
    paddingVertical: 15,
    paddingHorizontal: 15,
    justifyContent: "flex-start",
    backgroundColor: "rgba(255, 255, 255, 0.9)",
  },
  headingContainer: {
    flexDirection: "row",
    paddingBottom: 10,
    justifyContent: "space-between",
    borderBottomColor: "rgba(0, 1, 25, 0.1)",
    borderBottomWidth: 1,
  },
  heading: {
    color: "#000119",
    fontSize: 20,
    fontFamily: "roboto-condensed-bold",
  },
  buttonContainer: {
    paddingTop: 5,
    paddingHorizontal: 15,
    alignItems: "center",
  },
  details: {
    color: "#000119",
    fontSize: 14,
    fontFamily: "roboto-condensed-regular",
  },
  listContainer: {
    flex: 1,
  },
});

export default InfoCardStyles;
