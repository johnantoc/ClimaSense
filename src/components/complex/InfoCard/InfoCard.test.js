import React from 'react';
import InfoCard from './InfoCard';

import { cleanup, render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

afterEach(cleanup);

test('InfoCard renders correctly', () => {
  const tree = renderer.create(<InfoCard />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('InfoCard Details button click', () => {
  const { getByTestId } = render(<InfoCard />);
  fireEvent.press(getByTestId('btnGoToDetails'));
  expect(getByTestId('btnGoToDetails')).toBeDefined();
});
