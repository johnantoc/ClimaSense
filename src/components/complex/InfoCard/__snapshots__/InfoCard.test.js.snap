// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#000119",
        "flex": 0.6,
        "height": 100,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "borderTopLeftRadius": 80,
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "justifyContent": "center",
          },
        ]
      }
    >
      Hello World
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="btnGoToDetails"
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Details
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
